src/pineko/cli/fonll.py

```diff
@@ -69,83 +54,122 @@ def subcommand(
-
-    if ffns4til is None or ffns4bar is None:
-        raise InconsistentInputsError(
-            "At least one of ffns4til and ffns4bar should be provided."
-        )
-
-    # Do we consider two masses, i.e. mc and mb
-    if any([ffns5til, ffns5bar, ffn04]):
-        if (ffns5til is None and ffns5bar is None) or ffn04 is None:
-            raise InconsistentInputsError(
-                "To include nf5 contributions, ffn04 and at least one between ffns5til and ffns5bar are mandatory"
-            )
-
-    # Get theory info
-    tcard = theory_card.load(theoryid)
-    if tcard["DAMP"] == 1:
-        if not "DAMPPOWERc" in tcard or not "DAMPPOWERb" in tcard:
-            raise InconsistentInputsError(
-                "If DAMP is set, set also DAMPPOWERb and DAMPPOWERc"
-            )
-    else:
-        tcard["DAMPPOWERb"] = 0
-        tcard["DAMPPOWERc"] = 0
-    # Getting the paths to the grids
-    grids_name = grids_names(configs.configs["paths"]["ymldb"] / f"{dataset}.yaml")
-    for grid in grids_name:
-        # Checking if it already exists
-        new_fk_path = configs.configs["paths"]["fktables"] / str(theoryid) / grid
-        if new_fk_path.exists():
-            if not overwrite:
-                rich.print(
-                    f"[green]Success:[/] skipping existing FK Table {new_fk_path}"
-                )
-                return
-        fonll.produce_combined_fk(
-            *(
-                cfgpath(str(name), grid)
-                for name in (
-                    ffns3,
-                    ffn03,
-                    ffns4til,
-                    ffns4bar,
-                    ffn04,
-                    ffns5til,
-                    ffns5bar,
-                )
-            ),
-            theoryid,
-            damp=(tcard["DAMP"], tcard["DAMPPOWERc"], tcard["DAMPPOWERb"]),
-            cfg=cfg,
-        )
-        if new_fk_path.exists():
-            rich.print(f"[green]Success:[/] Wrote FK table to {new_fk_path}")
-        else:
-            rich.print(f"[red]Failure:[/]")
+    load_config(cfg)
+    fonll.assembly_combined_fk(
+        theoryid,
+        dataset,
+        ffns3,
+        ffn03,
+        ffns4til,
+        ffns4bar,
+        ffn04,
+        ffns5til,
+        ffns5bar,
+        overwrite,
+        cfg,
+    )
 
 
 @command.command("fonll_tcards")
 @click.argument("theoryID", type=int)
 @config_setting
 def fonll_tcards(theoryid, cfg):
     """Produce the FONLL tcards starting from the original tcard given by the theoryID."""
-    path = configs.detect(cfg)
-    base_configs = configs.load(path)
-    configs.configs = configs.defaults(base_configs)
+    load_config(cfg)
+
     tcard = theory_card.load(theoryid)
     tcard_parent_path = theory_card.path(theoryid).parent
     if "FONLL" not in tcard["FNS"]:
         raise TheoryCardError("The theorycard does not correspond to an FONLL scheme.")
     fonll.dump_tcards(tcard, tcard_parent_path, theoryid)
+
+
+@command.command("fonll_ekos")
```

Comment:
I'd say at this point we should definitely make a group (as for `theory`) and bind the subcommands there (thus also inverting `combine_fonll` to `fonll combine`)

Reply:
Yes, I agree that at this point it is worth doing this.

Reply:
okay done, although this will break everyone scripts.